### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by iskysun96**
Changes made
- added contributing guides
- edited README
- renamed subproject from `devportal-code-examples` to `python-contract-examples`

Request for @PhearZero :
- Review Contributing guides
- set up linting 
- set up CI/CD

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+```bash
+git clone git@github.com:algorandfoundation/devportal-code-examples.git
+```
+
+Update the latest changes from the remote repository
+
+```bash
+git fetch
+```
```
Comment (PhearZero) [inline CONTRIBUTING.md:25]: We could go further here and bring the developer all the way up with the project.

I would add:

- Starting localnet (`algokit localnet start`)
- Bootstrapping all of the projects for the first time (`algokit project bootstrap all`)
- Running the tests (`algokit project run test`)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+### Submitting Changes
+
+Provide a descriptive commit message that outlines the changes.
+Changes must be done via a Pull Request (PR).
+
+Ensure the following acceptance criteria are met before submitting a PR:
+
+- Ensure there are no typos in the content
+- Ensure `pnpm run lint` passes
```
Comment (PhearZero) [inline CONTRIBUTING.md]: These should generally match what is in the CI pipeline, basically an outline of what to expect.

- algokit project run audit
- algokit project run lint
- algokit project run test
- algokit project run build
- algokit project run audit-teal 

Aside: We could automate the checking locally with a git-hook but that is out of scope for now

**Review by PhearZero — COMMENTED** [review]
Looks good! I'm sure it will go through some refinements as the process develops and this is a great start!

File: CONTRIBUTING.md (hunk 2 above)
Comment (iskysun96) [reply CONTRIBUTING.md]: so are you saying we should include running all of these algokit commands in the guide?

**Review by PhearZero — APPROVED** [review]
LGTM 🕺!

**Comment by CiottiGiorgio** [comment]
Can we delete `initial-setup` branch now?

**Pull request merged**